### DISCUSSION TITLE
ResendRequest to replay all messages

### DISCUSSION THREAD

**Discussion by jacques0803** [quickfix-j/quickfixj#483] [General]
Hi FIX users,
We have a FIX Initiator app that only sends TradeCaptureReportRequest for trade date the current day, and then receive the TradeCaptureReports as they are executed on market.  

Lets say at the end of the day, I wish to manually issue a ResendRequest with BeginSeqNo=1 and EndSeqNo=0, i.e. all messages.  I can see from the log file the messages are retransmitted (received again)  with the PossibleDupFlag=Y, however, the Application::fromApp method does not seem to be invoked for these possible duplicate messages.

Now I understand, if QFJ has detected a loss of connection intra day, it would have resolved itself by issuing ResendRequests, so I have not actually missed receiving messages.  But what if something went wrong in the processing of the message (say tablespace full) and my storing of the message was not successful?  I want to generate a ResendRequest mainly to verify I have actually stored the message and was hoping the Application::toApp would be invoked again, even if a possible duplicate message.  Is this possible at all ?

Using:
FIX 5.0SP2
QFJ 2.3.1

Kind regards

**Comment by chrjohn** [comment]
Hi @jacques0803 ,

the general rule with this is that you should never interfere with the session level logic. QFJ will ignore the PossDup messages because their sequence is too low.
I would advise to persist the messages as you receive them so you can use them later and thus don't need to deal with the session logic.
If you *really* want to re-process the messages and ignore all warnings then you should probably set your target sequence number to 1 so that the messages are resent.

BTW, did I mention that you should never do session-level stuff manually? ;)

Cheers,
Chris.

**Reply by jacques0803** [reply]
Thanks for the comments @chrjohn.  You confirmed my gut feel: do not interfere with session level events.
